client/post-editor/editor-ground-control/index.jsx

```diff
@@ -1,113 +1,109 @@
 import AsyncLoad from 'components/async-load';
 import EditorPublishButton, { getPublishButtonStatus } from 'post-editor/editor-publish-button';
 import Button from 'components/button';
 import EditorPostType from 'post-editor/editor-post-type';
-import classNames from 'classnames';
 
-export default React.createClass( {
-	displayName: 'EditorGroundControl',
-
-	propTypes: {
-		hasContent: React.PropTypes.bool,
-		isConfirmationSidebarEnabled: React.PropTypes.bool,
-		isDirty: React.PropTypes.bool,
-		isSaveBlocked: React.PropTypes.bool,
-		isPublishing: React.PropTypes.bool,
-		isSaving: React.PropTypes.bool,
-		onPreview: React.PropTypes.func,
-		onPublish: React.PropTypes.func,
-		onSave: React.PropTypes.func,
-		onSaveDraft: React.PropTypes.func,
-		onMoreInfoAboutEmailVerify: React.PropTypes.func,
-		post: React.PropTypes.object,
-		savedPost: React.PropTypes.object,
-		setPostDate: React.PropTypes.func,
-		site: React.PropTypes.object,
-		user: React.PropTypes.object,
-		userUtils: React.PropTypes.object,
-		toggleSidebar: React.PropTypes.func,
-		type: React.PropTypes.string
-	},
-
-	mixins: [ PureRenderMixin ],
-
-	getDefaultProps: function() {
-		return {
-			hasContent: false,
-			isConfirmationSidebarEnabled: true,
-			isDirty: false,
-			isSaveBlocked: false,
-			isPublishing: false,
-			isSaving: false,
-			onPublish: noop,
-			onSaveDraft: noop,
-			post: null,
-			savedPost: null,
-			site: {},
-			user: null,
-			userUtils: null,
-			setPostDate: noop
-		};
-	},
-
-	componentDidMount: function() {
+export class EditorGroundControl extends PureComponent {
+	static propTypes = {
+		hasContent: PropTypes.bool,
+		isConfirmationSidebarEnabled: PropTypes.bool,
+		isDirty: PropTypes.bool,
+		isSaveBlocked: PropTypes.bool,
+		isPublishing: PropTypes.bool,
+		isSaving: PropTypes.bool,
+		moment: PropTypes.func,
+		onPreview: PropTypes.func,
+		onPublish: PropTypes.func,
+		onSave: PropTypes.func,
+		onSaveDraft: PropTypes.func,
+		onMoreInfoAboutEmailVerify: PropTypes.func,
+		post: PropTypes.object,
+		savedPost: PropTypes.object,
+		setPostDate: PropTypes.func,
+		site: PropTypes.object,
+		user: PropTypes.object,
+		userUtils: PropTypes.object,
+		toggleSidebar: PropTypes.func,
+		translate: PropTypes.func,
+		type: PropTypes.string
+	};
+
+	static defaultProps = {
+		hasContent: false,
+		isConfirmationSidebarEnabled: true,
+		isDirty: false,
+		isSaveBlocked: false,
+		isPublishing: false,
+		isSaving: false,
+		moment,
+		onPublish: noop,
+		onSaveDraft: noop,
+		post: null,
+		savedPost: null,
+		site: {},
+		translate: identity,
+		user: null,
+		userUtils: null,
+		setPostDate: noop
+	};
+
+	state = {
+		showSchedulePopover: false,
+		showAdvanceStatus: false,
+		firstDayOfTheMonth: this.getFirstDayOfTheMonth(),
+		lastDayOfTheMonth: this.getLastDayOfTheMonth(),
+		needsVerification: this.props.userUtils && this.props.userUtils.needsVerificationForSite( this.props.site ),
+	};
```

Comment:
This particular one results from applying the codemod, so whatever we decide, we should adapt the codemod accordingly. But TBH I wouldn't mind leaving the codemod as-is, and accepting its style as the norm 😛 

However, I had to cut-and-paste-after-running-the-codemod a couple of methods that the codemod would've otherwise choked on. I need to check if those are consistent with this one. Thanks for raising! 😄